test/controllers/api/v2/organizations_controller_test.rb

```diff
@@ -89,6 +89,95 @@ def test_create_duplicate_name
       assert_response :unprocessable_entity
     end
 
+    def test_create_with_auto_label
+      Organization.any_instance.stubs(:redhat_repository_url)
+      Organization.any_instance.stubs(:default_content_view).returns(OpenStruct.new(id: 1))
+      Organization.any_instance.stubs(:library).returns(OpenStruct.new(id: 10))
+      name = "Organization With Label"
+      assert_sync_task ::Actions::Katello::Organization::Create do |org|
+        org.name.must_equal name
+        assert org.valid?
+        org.label.wont_be_empty
+        org.stubs(:reload)
+      end
+      post :create, params: { :organization => {:name => name} }
+      assert_response :success
+    end
+
+    def test_create_identical_name_and_label
+      Organization.any_instance.stubs(:redhat_repository_url)
+      Organization.any_instance.stubs(:default_content_view).returns(OpenStruct.new(id: 1))
+      Organization.any_instance.stubs(:library).returns(OpenStruct.new(id: 10))
+      name = "organization_with_name_and_label_identical"
+      assert_sync_task ::Actions::Katello::Organization::Create do |org|
+        org.name.must_equal name
+        org.label.must_equal name
+        assert org.valid?
+        org.stubs(:reload)
+      end
+      post :create, params: { :organization => {:name => name, :label => name} }
+      assert_response :success
+    end
+
+    def test_create_with_name_and_label
+      Organization.any_instance.stubs(:redhat_repository_url)
+      Organization.any_instance.stubs(:default_content_view).returns(OpenStruct.new(id: 1))
+      Organization.any_instance.stubs(:library).returns(OpenStruct.new(id: 10))
+
+      name = "Organization With Label"
+      label = "org_with_label"
+      assert_sync_task ::Actions::Katello::Organization::Create do |org|
+        org.name.must_equal name
+        org.label.must_equal label
+        assert org.valid?
+        org.stubs(:reload)
+      end
+      post :create, params: { :organization => {:name => name, :label => label} }
+      assert_response :success
+    end
+
+    def test_create_with_name_label_description
+      Organization.any_instance.stubs(:redhat_repository_url)
+      Organization.any_instance.stubs(:default_content_view).returns(OpenStruct.new(id: 1))
+      Organization.any_instance.stubs(:library).returns(OpenStruct.new(id: 10))
+
+      name = "Organization With Name Label Description"
+      label = "organization_with_name_label_description"
+      description = "Organization Description"
+      assert_sync_task ::Actions::Katello::Organization::Create do |org|
+        org.name.must_equal name
+        org.label.must_equal label
+        org.description.must_equal description
+        assert org.valid?
+        org.stubs(:reload)
+      end
+      post :create, params: { :organization => {:name => name, :label => label, :description => description} }
+      assert_response :success
+    end
+
+    def test_create_with_name_description_auto_label
```

Comment:
what is the point of this test, since all functionality has already been checked in the previous tests?

Reply:
@JacobCallahan same as previous comment.
Any way if you feel that we are over testing things in robottelo it could be the right time to rise some issues in robottelo about deleting some test without porting them and we will include them in this port project to be deleted later when we will raise robottelo Delete test PR (but the delete of that tests from robottelo must be done in a stand alone PR along side with their issue).

Reply:
I'll respond to both comments here. I am all for removing duplicate functionality checks. Both in robottelo and this test base. The effort to scrub the robottelo tests probably isn't worth it, but porting things over here gives us a good chance on cutting down the tests that truly aren't necessary. 
We can have a discussion about this more during our Friday meeting. For now, I won't block this, If the katello team is fine with introducing potentially unnecessary tests.